### ISSUE TITLE
Add a setting to disable preview / view live functionality

### ISSUE THREAD

**Issue by bsnelder** [wagtail/wagtail#5670]
### Problem
I want to run Wagtail as a headless CMS. In my example, I only want to use Wagtail for a mobile app. This means I don't have a frontend to use the preview functionality for pages.

### Proposed solution
Introduce a new setting `WAGTAIL_HEADLESS_DISABLE_PREVIEW` that would disable the Preview / View live buttons for pages when set to `True`.

For now, there are alternatives like serving a page with some content saying Preview is not supported but more user-friendly would be to not show the buttons at all. Another option is to override some admin templates but this is also not ideal.

**Comment by kaedroho** [comment]
I would prefer to use the name `WAGTAILADMIN_DISABLE_PREVIEW` for consistency with other wagtail admin settings.

**Comment by gasman** [comment]
I'd be keen to look at alternatives to adding a new setting - if Wagtail isn't catering for a particular use-case then we should improve the design of Wagtail to accommodate it, not just add a new setting and conditional on top and call it done.

As @zerolab pointed out on Slack, it would make logical sense that setting `preview_modes` to an empty list should disable previewing - it looks like it's only failing because of the template-level logic. My preference would be to fix that behaviour.

**Comment by bsnelder** [comment]
> I'd be keen to look at alternatives to adding a new setting - if Wagtail isn't catering for a particular use-case then we should improve the design of Wagtail to accommodate it, not just add a new setting and conditional on top and call it done.
> 
> As @zerolab pointed out on Slack, it would make logical sense that setting `preview_modes` to an empty list should disable previewing - it looks like it's only failing because of the template-level logic. My preference would be to fix that behaviour.

Fixing the behavior of setting `preview_modes` to an empty list also makes sense. It then should not display a preview button at all. This is the first thing I tried before even asking the question on slack.

**Comment by gasman** [comment]
Fixed in #5781

**Comment by thibaudmartinez** [comment]
Hi there,

I am currently using Wagtail nightly `2.9.dev20200418` as I wanted to use the "disable preview" feature.

The live button is properly hidden, but I ran into the following problem.

Let's say I define the model SomePage for which I disable the preview.

````python
class SomePage(Page):
    preview_modes = []
````

I create a page from this model and save it has a draft. When I go to the admin home page and click on the button "DRAFT" of the row corresponding to the created page I get the following error.

````python
Internal Server Error: /admin/pages/4/view_draft/
Traceback (most recent call last):
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/wagtail/admin/urls/__init__.py", line 109, in wrapper
    return view_func(request, *args, **kwargs)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/wagtail/admin/auth.py", line 168, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/wagtail/admin/views/pages.py", line 610, in view_draft
    return page.make_preview_request(request, page.default_preview_mode)
  File "/home/thibaud/workspace/bugs/draft-btn/env/lib/python3.7/site-packages/wagtail/core/models.py", line 1398, in default_preview_mode
    return self.preview_modes[0][0]
IndexError: list index out of range
````
I also get this error when I click the "VIEW DRAFT" button of the page on the list page view.

It would be nice to hide these buttons for a page when `preview_modes` is set to an empty list.
